### PR TITLE
(#921) URI SAN records

### PR DESCRIPTION
**Pull request by vjanelle**
Support URI san records.

### CODE REVIEW

**Comment by vjanelle** [comment]
As kind of expected from #921  - Older versions of golang don't have URL support in the cypto/tls support.

**Comment by cbroglie** [comment]
Thanks @vjanelle! I don't see any reason to support more than the most recent 2 versions of Go, so you can go ahead and update the Travis configuration as part of this change. And would you mind updating the README as well?

File: csr/csr.go [hunk 1]
```diff
@@ -379,6 +383,9 @@ func Generate(priv crypto.Signer, req *CertificateRequest) (csr []byte, err erro
 			tpl.IPAddresses = append(tpl.IPAddresses, ip)
 		} else if email, err := mail.ParseAddress(req.Hosts[i]); err == nil && email != nil {
 			tpl.EmailAddresses = append(tpl.EmailAddresses, email.Address)
+		} else if uri, err := url.ParseRequestURI(req.Hosts[i]); err == nil && uri != nil {
+			log.Info("Found a URI: ", uri)
```
Comment (cbroglie) [inline csr/csr.go]: Please remove this log as well

Comment (vjanelle) [reply csr/csr.go]: oops.

**Comment by vjanelle** [comment]
@cbroglie done.

**Comment by vjanelle** [comment]
@cbroglie any opinions on if a squashed commit or not?

**Comment by cbroglie** [comment]
Separate commits is good here. Thanks again!

**Pull request merged**